### ISSUE TITLE
Support more DDS codecs

### ISSUE THREAD

**Issue by mikolalysenko** [mikolalysenko/regl#12]
Right now the DDS parsing code in regl is very minimal.  All of it can be found in the following file:

https://github.com/mikolalysenko/regl/blob/gh-pages/lib/util/parse-dds.js

But there is a lot more that DDS can do which is worth supporting.  In particular, having the ability to load high dynamic range (HDRI float) textures would be really cool, as well as supporting more compressed texture formats.

**Comment by mikolalysenko** [comment]
Info on the DDS format can be found on the MSDN here:  https://msdn.microsoft.com/en-us/library/windows/desktop/bb943991(v=vs.85).aspx

**Comment by vorg** [comment]
I've implemented the floating point cubemap parsing before https://github.com/Jam3/parse-dds

**Comment by mikolalysenko** [comment]
Awesome!  We should merge it in, though in the interest of minimizing external dependencies in regl it will probably have to get copied into the project directly.  @mattdesl are you ok with this?

**Comment by mattdesl** [comment]
Sure — but isn't that a bit counter to the standard node/npm approach? I think dependencies should be preferred for things like file parsing, otherwise regl will end up a monolith without any clear scope, not much different than ThreeJS or Babylon.

**Comment by mikolalysenko** [comment]
It's true, and it is a risk here.  However one of the design constraints that I set out with when building regl was to bring in no external dependencies.  This is something of an about face from things that I've tried in the past, but in order to just see this through I really want to try sticking to this constraint.

Maybe a better solution is to actually remove DDS parsing from the core of regl and instead rely on users to manually invoke parse-dds on their files before they load them.

I need to think it over a bit.
